sdk/include/opentelemetry/sdk/trace/span_data.h

```diff
@@ -84,6 +84,33 @@ struct AttributeConverter
   }
 };
 
+/**
+ * Class for storing events in SpanData.
+ */
+class SpanDataEvent
+{
+public:
+  SpanDataEvent(std::string name, core::SystemTimestamp timestamp)
+      : name_(name), timestamp_(timestamp)
+  {}
+
+  /**
+   * Get the name for this event
+   * @return the name for this event
+   */
+  std::string GetName() const noexcept { return name_; }
```

Comment:
std::string ->  -> opentelemetry::nostd::string_view (for consistency with how SpanData returns string members, and it also should avoid copy).